content/asm_4.md

```diff
@@ -1,194 +1,303 @@
 
 ```assembly
-section .bss
-		OUTPUT resb 12
+cmp rax, rdx
+cmovne rax, rdx
 ```
 
-Ok we have some data and buffer where to put result, now we can define text section for code. Let's start from main _start routine:
+## Binary arithmetic instructions
 
-```assembly
-_start:
-		mov rsi, INPUT
-		xor rcx, rcx
-		cld
-		mov rdi, $ + 15
-		call calculateStrLength
-		xor rax, rax
-		xor rdi, rdi
-		jmp reverseStr
-```
+We use the decimal arithmetic instructions to operate with integer numbers. We already have seen these instructions in one of the previous chapters:
+
+- `add`  - Addition. Add the value of the second operand to the first and store the result in the first operand.
+- `sub`  - Subtraction. Subtract the value of the second operand from the first and store the result in the first operand.
+- `mul` and `imul`  - Unsigned and signed multiplication. Both instructions takes a single operand. The value of this operand is multiplied to the value of the `rax` register and the result is stored in the pair of the `rax` and `rdx` registers. The multiplication of two `n` bit numbers may lead to the `2 * n` result. That is the reason why the result is stored in the `rdx:rax` pair.
+- `div` and `idiv`  - Unsigned and signed division. Both instructions takes a single operand. The value of this operand is divided on the value of the `rax` register. The place where the result is stored depends on the size of operands. In a case of 8-bit operands, the result is stored in the `al:ah` pair. In a case of `16-bit` operands the result is stored in the `dx:ax` pair, for `32-bit` operands the result is stored in the `edx:eax`, and in the case of `64` bit operands the result is stored in the `rdx:rax` pair.
+- `inc`  - Increment the value of the first operand.
+- `dec`  - Decrement the value of the first operand.
+- `neg`  - Negate the value of the first operand.
 
-Here are some new things. Let's see how it works: First of all we put INPUT address to si register at line 2, as we did for writing to stdout and write zeros to rcx register, it will be counter for calculating length of our string. At line 4 we can see cld operator. It resets df flag to zero. We need in it because when we will calculate length of string, we will go through symbols of this string, and if df flag will be 0, we will handle symbols of string from left to right. Next we call calculateStrLength function. I missed line 5 with mov rdi, $ + 15 instruction, i will tell about it little later. And now let's look at calculateStrLength implementation:
+For example:
 
 ```assembly
-calculateStrLength:
-		;; check is it end of string
-		cmp byte [rsi], 0
-		;; if yes exit from function
-		je exitFromRoutine
-		;; load byte from rsi to al and inc rsi
-		lodsb
-		;; push symbol to stack
-		push rax
-		;; increase counter
-		inc rcx
-		;; loop again
-		jmp calculateStrLength
+;; Increment the value of the rcx
+inc rcx
+
+;; Add the value of the rcx to rdx
```

Comment:
```suggestion
;; Add the value of rcx to rdx
```

Reply:
Not sure: incremets/adds (as we are talking about "this instruction that DOES these actions"), or increment/add as just an action 🤔

Reply:
I'd leave just an action